### PR TITLE
tsdb: add enable overlapping compaction

### PR DESCRIPTION
**Pull request by GiedriusS**
This functionality is needed in downstream projects because they have a separate component that does compaction.

Upstreaming
https://github.com/grafana/mimir-prometheus/blob/7c8e9a2a76fc729e9078889782928b2fdfe240e9/tsdb/compact.go#L323-L325.

### CODE REVIEW

File: tsdb/compact.go [hunk 1]
```diff
@@ -191,14 +195,15 @@ func NewLeveledCompactorWithOptions(ctx context.Context, r prometheus.Registerer
...
-		postingsEncoder:          pe,
+		ranges:                       ranges,
+		chunkPool:                    pool,
+		logger:                       l,
+		metrics:                      newCompactorMetrics(r),
+		ctx:                          ctx,
+		maxBlockChunkSegmentSize:     maxBlockChunkSegmentSize,
+		mergeFunc:                    mergeFunc,
+		postingsEncoder:              pe,
+		disableOverlappingCompaction: opts.DisableOverlappingCompaction,
```
Comment (SuperQ) [inline tsdb/compact.go:206]: We prefer positive flags, so this should be `enableOverlappingCompaction`.

Comment (GiedriusS) [reply tsdb/compact.go]: :bow: thanks for the review, I've updated it to be `enableOverlappingCompaction`.

**Review by SuperQ — APPROVED** [review]

**Review by pracucci — APPROVED** [review]
Thanks! I was going to open the same PR too, cause we're using it in Mimir.

**Review by gotjosh — APPROVED** [review]
LGTM, seems safe enough.

**Comment by bboreham** [comment]
Previous change in #13282

**Comment by GiedriusS** [comment]
Yes, sorry for the churn. I didn't realize what this flag was for until I went down the rabbit hole.

**Comment by bboreham** [comment]
Is it worth addding a note for the next person?

The previous note (which was not sufficient) read:

```
	// The reason why this flag exists is because there are various users of the TSDB
	// that do not want vertical compaction happening on ingest time. Instead,
	// they'd rather keep overlapping blocks and let another component do the overlapping compaction later.
```

**Pull request merged**